### PR TITLE
faxsms module cleanup and implement send email client

### LINKED ISSUE #7859: Refactor oe-module-faxsms abstract base class and clients
### Describe the bug

After new Clickatell client addition, some worrisome credentials and some base methods need to be refactored for use in all other previous clients.
This is more or less a clean up of older workflows where some ambiguities' occur and duplicates common workflow across all clients.
This will make it easier for new dev's adding new vendors based on the Clickatell authors experience.

### Client configuration

- Browser:
- OpenEMR version: 7.0.2.3
- Operating system:

### LINKED ISSUE #7865: Need universal compose and send email for practice
### Describe the feature
It's nice to be able to send a quick email from strategic locations throughout the U.I..
Starting from the new Email Module and i'm looking for somewhere in Patient Dashboard.
The new dialogs have the ability to send both email and SMS depending on what modules are enabled e.g. Secure Messaging.

Having asked several user lately with all agreeing this would be very useful plus i've had the request several time in the past few years.
Eventually I'll log and track mail sent and report in the email module. No plans to receive yet but maybe.
Feature is event based with several dialog feature dynamic via event data. Very interest for any feedback.

- From Module

image

- Note the Address book and Patient lookup helper popups.

image

- Notify for Portal Document with onetime token to document in question.
- 
image

- And Secure Messaging

image

Most likely will have PR up tomorrow along with some reorg of dispatch abstract class and am going to add searchable comment tags in code showing where new client data items are added to workflow.

While my original design goal was to have the notification/communication module support all vendors and types(SMS, fax and email) from one module, admittedly; I worried if I could reliably pull it off. But now that we have four vendors with AWS Pinpoint up next everything seems to be running smooth at least on minor traffic.

Anyway hope folks will take out for a test drive after merge.
BTW: will support two SMTP accounts from with to choose.

### LINKED ISSUE #7867: Deleting messages in the Message Center
### Describe the bug
When trying to click on a checkbox to delete a message from the message center, it automatically opens the message.
Make it possible to select the checkbox without opening the messages, so we can delete the messages that are done?

### To Reproduce

1. Add a '....'

### Client configuration

- Browser:
- OpenEMR version:7.0.2 dev
- Operating system:

### PR DESCRIPTION
**Pull request by sjpadgett**
Fixes #7865 
Fixes #7859 
Fixes #7867 

#### Changes proposed in this pull request:
- Reorg dispatch class for better clarification
- Add event handlers for SMS and Email buttons and listeners.
- Remove the temporary vendor directory to support RingCentral patch. 
- Dependencies are now and have been included in core composer.
- Fix message center delete checkbox from opening message on select.
- 
#### Does your code include anything generated by an AI Engine? No

If anyone wants to review that'd be fine as I think I'm close.

### CODE REVIEW

**Comment by sjpadgett** [comment]
NOTE for me: Add a Subject text input for emails.

File: interface/main/calendar/find_patient_popup.php [hunk 1]
```diff
@@ -46,239 +46,254 @@
...
+                                          } ?>><?php echo xlt('ID'); ?></option>
+                        <option value="SSN"<?php if (!empty($searchby) && ($searchby == 'SSN')) {
+                            echo ' selected';
+                                           } ?>><?php echo xlt('SSN'); ?></option>
+                        <option value="DOB"<?php if (!empty($searchby) && ($searchby == 'DOB')) {
+                            echo ' selected';
+                                           } ?>><?php echo xlt('DOB'); ?></option>
+                    </select>
+                    <label for="searchparm" class="col-form-label col-form-label-sm col"><?php echo text(xl('for:')); ?></label>
+                    <input type='text' class="form-control form-control-sm col" id='searchparm' name='searchparm' size='12' value='<?php echo attr($_REQUEST['searchparm'] ?? ''); ?>' title='<?php echo text(xl('If name, any part of lastname or lastname,firstname'), ENT_QUOTES); ?>' />
```
Comment (bradymiller) [inline interface/main/calendar/find_patient_popup.php:148]: nice to see going to the abbreviated functions in this script :)
Have a left over `ENT_QUOTES` above

Comment (bradymiller) [reply interface/main/calendar/find_patient_popup.php:148]: actually and should be attr() or could go to xla() if drop the xl()

File: interface/main/calendar/find_patient_popup.php [hunk 2]
```diff
@@ -46,239 +46,254 @@
...
+                    <div class="col">
+                        <input class='btn btn-primary btn-sm' type='submit' id="submitbtn" value='<?php echo text(xl('Search'), ENT_QUOTES); ?>' />
```
Comment (bradymiller) [inline interface/main/calendar/find_patient_popup.php:150]: Have a left over `ENT_QUOTES` above
and it should be attr() or could go to xla() if drop the xl()

File: interface/main/calendar/find_patient_popup.php [hunk 3]
```diff
@@ -46,239 +46,254 @@
...
 
-// show the 'searching...' status and submit the form
-var SubmitForm = function(eObj) {
-    $("#submitbtn").css("disabled", "true");
-    $("#searchspinner").css("visibility", "visible");
-    return true;
-}
+            function selpid(pid, lname, fname, dob) {
+                if (opener.closed || !opener.setpatient)
+                    alert("<?php echo text(xl('The destination form was closed; I cannot act on your selection.'), ENT_QUOTES); ?>");
```
Comment (bradymiller) [inline interface/main/calendar/find_patient_popup.php:272]: Have a left over ENT_QUOTES above
and it should be attr() or could go to xla() if drop the xl()
even better go to xlj (and drop outside quotes)

File: interface/modules/custom_modules/oe-module-faxsms/src/Events/NotificationEventListener.php [hunk 4]
```diff
@@ -114,17 +115,21 @@ public function onNotifyDocumentRenderOneTime(SendNotificationEvent $event)
...
         if (
             !empty($recipientEmail)
             && ($includeEmail)
             && ($patient['hipaa_allowemail'] == 'YES')
         ) {
+            $status .= "Sending email to $recipientEmail" . ': ';
             $status .= $this->emailNotification($recipientEmail, $html_message);
         }
+        $status .= "\n";
+        echo (nl2br($status));
```
Comment (bradymiller) [inline interface/modules/custom_modules/oe-module-faxsms/src/Events/NotificationEventListener.php:132]: looks likes $status should be escaped

**Pull request merged**